JavaScript/JavaScriptSDK/Context/User.ts

```diff
@@ -29,12 +33,60 @@ module Microsoft.ApplicationInsights.Context {
          * The store region.
          */
         public storeRegion: string;
-        
+
+        /**
+         * Authorized user id
+         */
+        private authId: string;
+        public getAuthId() {
+            return this.authId;
+        }
+
+        /**
+         * Sets the autheticated user id and the account id in this session.
+         * @param id {string} - The autheticated user id
+         * @param accountId {string} - The account id.
+         * @returns {} 
+         */
+        public setAuthUserContext(id: string, accountId?: string) {
+
+            // Validate inputs to ensure no cookie control characters.
+            var isInvalidInput = !this.validateUserInput(id) || (accountId && !this.validateUserInput(accountId));
+            if (isInvalidInput) {
+                _InternalLogging.throwInternalUserActionable(LoggingSeverity.WARNING, "Setting auth user context failed. " +
+                    "User auth/account id should be of type string, and not contain commas, semi-colons, equal signs, spaces, or vertical-bars.");
+                return;
+            }
+
+            // Create cookie string.
+            this.authId = id;
+            var authCookie = this.authId;
+            if (accountId) {
+                this.accountId = accountId;
+                authCookie = [this.authId, this.accountId].join(User.cookieSeparator);
+            }
+            
+            // Set the cookie. No expiration date because this is a session cookie (expires when browser closed).
+            // Encoding the cookie to handle unexpected unicode characters.
+            Util.setCookie(User.authUserCookieName, encodeURI(authCookie));
+        }
+
+        /**
+         * Clears the authenticated user id and the account id from the user context.
+         * @returns {} 
+         */
+        public clearAuthUserContext() {
```

Comment:
I dont see a need for this currently. Generally speaking, we want to call set on log-in, and clear on log-off. Clearing just one of the values would be a "partial" log-off scenario.

We do support scenarios where in 1 session the same user switches between accounts, or switching users within the same account - it's possible to just call set again with the new context.